aws/rust-runtime/aws-inlineable/src/s3_express.rs

```diff
@@ -103,23 +103,383 @@ pub(crate) mod auth {
 
 /// Supporting code for S3 Express identity cache
 pub(crate) mod identity_cache {
+    use aws_credential_types::Credentials;
+    use aws_smithy_async::time::SharedTimeSource;
+    use aws_smithy_runtime::expiring_cache::ExpiringCache;
+    use aws_smithy_runtime_api::box_error::BoxError;
+    use aws_smithy_runtime_api::client::identity::Identity;
+    use hmac::{digest::FixedOutput, Hmac, Mac};
+    use lru::LruCache;
+    use sha2::Sha256;
+    use std::fmt;
+    use std::future::Future;
+    use std::hash::Hash;
+    use std::sync::Mutex;
+    use std::time::{Duration, SystemTime};
+
+    pub(crate) const DEFAULT_MAX_CACHE_CAPACITY: usize = 100;
+    pub(crate) const DEFAULT_BUFFER_TIME: Duration = Duration::from_secs(10);
+
+    #[derive(Clone, Eq, PartialEq, Hash)]
+    pub(crate) struct CacheKey(String);
+
+    impl CacheKey {
+        pub(crate) fn from(bucket_name: &str, creds: &Credentials) -> Self {
```

Comment:
I'd probably name this `new` since it's not a `from` implementation

Reply:
Addressed in 5dcaa96.